iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -127,14 +133,23 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
           .run(output -> {
             Table table = HiveIcebergStorageHandler.table(context.getJobConf(), output);
             if (table != null) {
-              HiveIcebergWriter writer = writers.get(output);
+              Collection<DataFile> dataFiles = Lists.newArrayList();
+              Collection<DeleteFile> deleteFiles = Lists.newArrayList();
               String fileForCommitLocation = generateFileForCommitLocation(table.location(), jobConf,
-                  attemptID.getJobID(), attemptID.getTaskID().getId());
-              if (writer != null) {
-                createFileForCommit(writer.files(), fileForCommitLocation, table.io());
-              } else {
+                      attemptID.getJobID(), attemptID.getTaskID().getId());
+              if (writers.get(output) != null) {
+                for (HiveIcebergWriter writer : writers.get(output)) {
+                  if (writer != null) {
+                    dataFiles.addAll(writer.files().dataFiles());
+                    deleteFiles.addAll(writer.files().deleteFiles());
```

Comment:
`writer.files()` might create an object every time. We should keep the object and reuse it.

Reply:
Reverted this.